source/posix/environment.c

```diff
@@ -14,7 +14,7 @@ int aws_get_environment_value(
     struct aws_string **value_out) {
 
     const char *value = getenv(aws_string_c_str(variable_name));
-    if (value == NULL) {
+    if (value == NULL || value[0] == '\0') {
```

Comment:
It sucks, but this is how it is with environment variables. Sometimes just the existence of the env-var is meaningful, even if it's blank. Maybe create an alternate function like `aws_get_environment_variable_nonempty()` that has this behavior, and scrub the codebase for places that would benefit from using that instead

Reply:
Yeah, I agree. We can discuss the following options in the office tomorrow.

1. Modify the existing function, and we can create a new function in the future if required, like if we have a use case where `env-var` can be blank.

2. Create a new function, `aws_get_environment_variable_nonempty`, `aws_get_environment_variable_cursor`, or `aws_get_environment_variable_c_str`. I prefer the `aws_get_environment_variable_cursor` if we decide to just create a new function.